airbyte-webapp/src/views/Connection/ConnectionForm/components/ScheduleField.tsx

```diff
@@ -0,0 +1,170 @@
+        basicSchedule: undefined,
+      };
+    } else {
+      scheduleData = {
+        basicSchedule: item.value,
+      };
+    }
+
+    form.setFieldValue(field.name, scheduleData, true);
+
+    // Also set scheduleType for yup validation
+    const scheduleType = isManualOrCron ? (item.value as ConnectionScheduleType) : ConnectionScheduleType.basic;
+    form.setFieldValue("scheduleType", scheduleType, true);
+  };
+
+  const getBasicScheduleValue = (value: ConnectionScheduleData, form: FormikProps<FormikConnectionFormValues>) => {
+    const scheduleType = form.values.scheduleType;
+
+    if (scheduleType === ConnectionScheduleType.basic) {
+      return value.basicSchedule;
+    }
+
+    return formatMessage({
+      id: `frequency.${scheduleType}`,
+    }).toLowerCase();
+  };
+
+  const getZoneValue = (currentSelectedZone: string | undefined): string => {
+    if (!currentSelectedZone) {
+      return cronTimeZones[0].value;
+    }
+
+    return currentSelectedZone;
+  };
+
+  const onCronChange = (
+    event: DropDownRow.IDataItem | ChangeEvent<HTMLInputElement>,
+    field: FieldInputProps<ConnectionScheduleData>,
+    form: FormikProps<FormikConnectionFormValues>,
+    key: string
+  ) => {
+    form.setFieldValue(field.name, {
+      cron: {
+        ...field.value?.cron,
+        [key]: (event as DropDownRow.IDataItem).value
+          ? (event as DropDownRow.IDataItem).value
+          : (event as ChangeEvent<HTMLInputElement>).currentTarget.value,
+      },
+    });
+  };
+
+  const cronTimeZones = useMemo(() => {
+    return availableCronTimeZones.map((zone: string) => ({ label: zone, value: zone }));
+  }, []);
+
+  const isCron = (form: FormikProps<FormikConnectionFormValues>): boolean => {
+    return form.values.scheduleType === ConnectionScheduleType.cron;
+  };
+
+  return (
+    <Field name="scheduleData">
+      {({ field, meta, form }: FieldProps<ConnectionScheduleData>) => (
+        <>
+          <div className={styles.flexRow}>
+            <div className={styles.leftFieldCol}>
+              <ControlLabels
+                className={styles.connectorLabel}
+                nextLine
+                error={!!meta.error && meta.touched}
+                label={formatMessage({
+                  id: "form.frequency",
+                })}
+                message={formatMessage({
+                  id: "form.frequency.message",
+                })}
+              />
+            </div>
+            <div className={styles.rightFieldCol} style={{ pointerEvents: mode === "readonly" ? "none" : "auto" }}>
+              <DropDown
+                {...field}
+                error={!!meta.error && meta.touched}
+                options={frequencies}
+                onChange={(item) => {
+                  onScheduleChange(item, field, form);
+                }}
+                value={getBasicScheduleValue(field.value, form)}
+              />
+            </div>
+          </div>
+          {isCron(form) && (
```

Comment:
Another bug I am seeing when testing this locally: when I select `Cron` from the dropdown, it immediately says `The form is invalid` at the bottom, and this doesn't go away until I select a different timezone.

This is not expected -- we shouldn't say that the form is invalid until the user has typed in some cron value and then clicked away from that input.

I suspect that this is because `setFieldValue` is not being called with a proper timezone until the user has actually selected a value from the dropdown. Could you also investigate this more?

And related to this - we may want to set have something like UTC or Pacific time be the default value that is selected when the Cron expression inputs first pop up, instead of `Africa/Abidjan`, since there are not likely to be a lot of people choosing that timezone.

Reply:
Yeah it says that because we are validating it after setting the value for `scheduleData`. Even though we don't validate then it automatically validates after blur. 

Thus I have defaulted the cron
```
const CRON_DEFAULT_VALUE = {
  cronTimeZone: "UTC",
  // Fire at 12:00 PM (noon) every day
  cronExpression: "0 0 12 * * ?",
};
```

I thought this will be a better solution. Kindly suggest if something is off or if there is another way to do it.

Reply:
I think a default value like this is a helpful starting point, and also solves the invalid form problem 👍